data/json/techniques.json

```
@@ -260,6 +260,78 @@
     "defensive": true,
     "grab_break": true
   },
+  {
+    "type": "technique",
+    "id": "tec_bojutsu_thrust",
+    "name": "Front Thrust",
+    "messages": [ "You thrust at the %s", "<npcname> thrusts at %s" ],
+    "skill_requirements": [ { "name": "melee", "level": 1 } ],
+    "melee_allowed": true,
+    "mult_bonuses": [ { "stat": "damage", "type": "bash", "scale": 1.2 } ]
+  },
+  {
+    "type": "technique",
+    "id": "tec_bojutsu_kneestrike",
+    "name": "Knee Strike",
+    "messages": [
+      "After blocking %s's attack you quickly strike at their knee!",
+      "After blocking %s's attack <npcname> quickly strikes at their knee!"
+    ],
+    "skill_requirements": [ { "name": "melee", "level": 2 } ],
+    "melee_allowed": true,
+    "block_counter": true,
+    "crit_ok": true,
+    "down_dur": 1,
+    "mult_bonuses": [ { "stat": "movecost", "scale": 0.8 } ]
+  },
+  {
+    "type": "technique",
+    "id": "tec_bojutsu_frontstrike",
+    "name": "Overhead Front Strike",
+    "messages": [
+      "You block %s's attack and counter with a Overhead Front Strike!",
+      "<npcname> blocks %s's attack and counter with a Overhead Front Strike!"
+    ],
+    "skill_requirements": [ { "name": "melee", "level": 3 } ],
+    "melee_allowed": true,
+    "block_counter": true,
+    "crit_ok": true,
+    "stun_dur": 1,
+    "mult_bonuses": [ { "stat": "damage", "type": "bash", "scale": 1.4 } ]
+  },
+  {
+    "type": "technique",
+    "id": "tec_bojutsu_rollingstrike",
+    "name": "Rolling Strike",
+    "messages": [ "Using your rolling staff you quickly strike %s", "Using rolling staff <npcname> quickly strikes %s" ],
+    "skill_requirements": [ { "name": "melee", "level": 3 } ],
+    "melee_allowed": true,
+    "req_buffs": [ "buff_bojutsu_onmove" ],
```

```suggestion
    "required_buffs_all": [ "buff_bojutsu_onmove" ],
```
dunno is it what it need